csrc/multidevice/executor.cpp

```diff
@@ -0,0 +1,140 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <device_lower/utils.h>
+#include <fusion_segmenter.h>
+#include <ir/utils.h>
+#include <multidevice/executor.h>
+
+namespace nvfuser {
+
+bool PipelineExecutor::shouldRun(PipelineStage* stage) {
+  should_run_.try_emplace(
+      stage,
+      std::count(
+          stage->descriptor()->mesh.deviceIndices().begin(),
+          stage->descriptor()->mesh.deviceIndices().end(),
+          runtime_.rankToDeviceIdx(runtime_.comm_.rank())));
+  return should_run_[stage];
+}
+
+void PipelineExecutor::handle(PipelineStage* stage) {
+  // get the IValues corresponding to the stage's input
+  std::vector<c10::IValue> stage_input_IValues;
+  for (auto& input_val : stage->inputs()) {
+    stage_input_IValues.push_back(val_to_IValue_[input_val]);
+  }
+
+  // Create the stage executor
+  fec_.try_emplace(
```

Comment:
Similar to https://github.com/NVIDIA/Fuser/pull/483#discussion_r1253683844, I don't think this implement the desired behavior.

Reply:
see https://github.com/NVIDIA/Fuser/pull/483#discussion_r1254298504